protocol/bc/map.go

```diff
@@ -68,14 +69,20 @@ func mapTx(tx *TxData) (headerID Hash, hdr *TxHeader, entryMap map[Hash]Entry, e
 			// programs are omitted here because they do not contribute to
 			// the body hash of an issuance.
 
-			var nonce Entry
+			var (
+				anchor      Entry
+				setAnchored func(Hash, *Issuance)
+			)
 
 			if len(oldIss.Nonce) == 0 {
 				if firstSpend == nil {
 					err = fmt.Errorf("nonce-less issuance in transaction with no spends")
 					return
 				}
-				nonce = firstSpend
+				anchor = firstSpend
+				setAnchored = func(id Hash, iss *Issuance) {
```

Comment:
This could be `setAnchored = firstSpend.SetAnchored` if you want. Same for nonce below.